tracer/test/test-applications/integrations/Samples.GraphQL7/StarWars/StarWarsData.cs

```diff
@@ -0,0 +1,88 @@
+using System;
```

Comment:
(I'm assume the code the `StarWars` fold is exactly the same across all three samples, if it's not then don't worry about this).

Reply:
The code has been migrated to support the last version and so the code is slightly changed in some classes.

Reply:
I tried to merge the 2 samples beginning but I faced one problem.
When I create a new test specified for a newer version, I need to update the version of the `ApiVersion` in the `.csproj` sample file. However, if I start the tests on an older version, the default version selected to run the test is the version specified in the `.csproj`. It seems that I can't enforce the version by test because in the `PackagesVersions` definition, when tests are run with filters, the flag `DEFAULT_SAMPLES` is set, and the versions list is set to an empty string (cf: `PackageVersionsLatestMinors.g.cs`)

Reply:
You can try it on this separate branch `graphql-version-upgrade-flavien-test-merge-sample` https://github.com/DataDog/dd-trace-dotnet/tree/graphql-version-upgrade-flavien-test-merge-sample
The default version used for building the GraphQL tests is `7.3.0` as written in the `.csproj` even if the test GraphQL4Tests is run.
`./build.sh BuildAndRunLinuxIntegrationTests --framework "net7.0" --filter "graphql4" --SampleName "Samples.GraphQL4"`

Reply:
The other branch looks not too bad. I think one issue is that you add a new entry in `PackageVersionsGeneratorDefinitions.json` whereas you could just change the versions as you use the same sample.

Reply:
To close this conversation:
After discussing with @andrewlock, we come up with a middle ground approach. Use `#ifdef` in graphql4 sample, create a separate graphql7 sample and include files of graphql4 to it.
So the sample graphql7 is kept and the tests running on older versions runs with no issue.